classes/steps/actions/webservice_action_step.php

```diff
@@ -0,0 +1,162 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_trigger\steps\actions;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once($CFG->libdir . '/externallib.php');
+
+/**
+ * Webservice action step class.
+ *
+ * @package    tool_trigger
+ * @author     Kevin Pham <kevinpham@catalyst-au.net>
+ * @copyright  Catalyst IT, 2022
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class webservice_action_step extends base_action_step {
+
+    use \tool_trigger\helper\datafield_manager;
+
+    /** @var string $functionname Name of the function to be called */
+    protected $functionname;
+
+    /** @var int $username The user in which this action will be actioned in the context of */
+    protected $username;
+
+    /** @var int $params parameters that will be used in the corresponding web service function call */
+    protected $params;
+
+    /**
+     * The fields supplied by this step.
+     *
+     * @var array
+     */
+    private static $stepfields = [
+        'data',
+        'error',
+        'exception',
+    ];
+
+    protected function init() {
+        $this->functionname = $this->data['functionname'];
+        $this->username = $this->data['username'];
+        $this->params = $this->data['params'];
+    }
+
+    /**
+     * Returns the step name.
+     *
+     * @return string human readable step name.
+     */
+    public static function get_step_name() {
+        return get_string('webserviceactionstepname', 'tool_trigger');
+    }
+
+    /**
+     * Returns the step name.
+     *
+     * @return string human readable step name.
+     */
+    public static function get_step_desc() {
+        return get_string('webserviceactionstepdesc', 'tool_trigger');
+    }
+
+    /**
+     * Runs the configured step.
+     *
+     * @param $trigger
+     * @param $event
+     * @param $stepresults - result of previousstep to include in processing this step.
+     * @return array if execution was succesful and the response from the execution.
+     */
+    public function execute($step, $trigger, $event, $stepresults) {
+        global $CFG, $DB;
+        require_once($CFG->dirroot . '/admin/tool/trigger/guzzle/autoloader.php');
```

Comment:
why do you need guzzle?

Reply:
Remnant code from the HTTP post handling copy. Will remove.